### PR TITLE
Formplayer Case Tiles

### PR DESCRIPTION
**Pull request by wpride**
image

Remake of https://github.com/dimagi/formplayer/pull/47

cross https://github.com/dimagi/commcare-hq/pull/12281

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -81,6 +81,12 @@ dependencies {
     testCompile("com.jayway.jsonpath:json-path-assert:0.8.1")
 }
 
+test {
+    testLogging {
+        exceptionFormat = 'full'
```
Comment (wpride) [inline build.gradle]: @benrudolph so we'll have the stack trace on Travis

Comment (benrudolph) [inline build.gradle]: 🙌

File: src/main/java/beans/menus/Style.java [hunk 2]
```diff
@@ -18,7 +18,7 @@ public Style(DetailField detail){
             try {
                 setFontSize(Integer.parseInt(detail.getFontSize()));
             } catch (NumberFormatException nfe) {
-                nfe.printStackTrace();
+                //nfe.printStackTrace();
```
Comment (benrudolph) [inline src/main/java/beans/menus/Style.java]: something should be handled here or logging of the stack trace. maybe set a default fontsize?

File: src/main/java/beans/menus/Tile.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+    public int getGridHeight() {
+        return gridHeight;
+    }
+
+    public void setGridHeight(int gridHeight) {
+        this.gridHeight = gridHeight;
+    }
+
+    public String getCssId() {
```
Comment (benrudolph) [inline src/main/java/beans/menus/Tile.java]: it's kind of strange that the java class knows about css

File: src/test/java/tests/BaseMenuTestClass.java [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+                            engine.initFromLocalFileResource(absolutePath);
+                        } else if (absolutePath.endsWith(".ccz")) {
+                            engine.initFromArchive(absolutePath);
+                        } else {
+                            throw new RuntimeException("Can't install with reference: " + absolutePath);
+                        }
+                        engine.initEnvironment();
+                        return engine;
+                    } catch (Exception e) {
+                        e.printStackTrace();
```
Comment (benrudolph) [inline src/test/java/tests/BaseMenuTestClass.java]: doesn't hte stack trace get printed and halt execution anyways? seems like it'd have the same behavior if we removed this try block

File: src/main/java/beans/menus/Tile.java (hunk 3 above)
Comment (wpride) [inline src/main/java/beans/menus/Tile.java]: Its legacy from mWellCare unfortunately, seemed like a good idea at the time :(

**Comment by benrudolph** [comment]
👍 on ✔️

**Pull request merged**